cli/cmd/cluster_prepare.go

```diff
@@ -171,42 +182,42 @@ func (r *runners) prepareCluster(_ *cobra.Command, args []string) error {
 	// wait for the wait group
 	wg.Wait()
 
-	// get the kubeconfig
-	kubeconfig, err := r.kotsAPI.GetClusterKubeconfig(clusterID)
-	if err != nil {
-		return errors.Wrap(err, "get cluster kubeconfig")
-	}
-
+	log.ActionWithSpinner("Waiting for release to be ready")
 	isReleaseReady, err := isReleaseReadyToInstall(r, log, *release)
 	if err != nil || !isReleaseReady {
+		log.FinishSpinnerWithError()
 		return errors.Wrap(err, "release not ready")
 	}
+	log.FinishSpinner()
 
-	// run preflights
-
-	// install the chart or application
-	for _, chart := range release.Charts {
-		output, err := installChartRelease(a.Slug, release.Sequence, chart.Name, email, customer.InstallationID, kubeconfig, r.args.prepareClusterValuesPath, r.args.prepareClusterValueItems)
-		if err != nil {
-			return errors.Wrap(err, "install release")
-		}
+	kubeconfig, err := r.kotsAPI.GetClusterKubeconfig(clusterID)
+	if err != nil {
+		return errors.Wrap(err, "get cluster kubeconfig")
+	}
 
-		// print the output
-		fmt.Fprintf(r.w, "%s\n", output)
+	kubeconfigFile, err := os.CreateTemp("", "kubeconfig")
+	if err != nil {
+		return errors.Wrap(err, "create kubeconfig file")
+	}
+	defer os.Remove(kubeconfigFile.Name())
+	if err := os.WriteFile(kubeconfigFile.Name(), kubeconfig, 0644); err != nil {
```

Comment:
kubeconfigFile is an os.File. should you write to it rather than using os.WriteFile?

Reply:
made changes to use os.File.Write